feat: Make library local to given excalidraw instance and allow consumer to control it (#3451)
excalidraw/excalidraw: src/packages/excalidraw/CHANGELOG.md

### Features

> - Don't share library & attach to the excalidraw instance [#3451](https://github.com/excalidraw/excalidraw/pull/3451).

Reviewer: ```suggestion
- Make library local to given excalidraw instance (previously, all instances on the same page shared one global library) [#3451](https://github.com/excalidraw/excalidraw/pull/3451).
```

  - Added prop [onLibraryChange](https://github.com/excalidraw/excalidraw/blob/master/src/packages/excalidraw/README.md#onLibraryChange) which if supplied will be called when library is updated.
  - Added attribute `libraryItems` to prop [initialData](https://github.com/excalidraw/excalidraw/blob/master/src/packages/excalidraw/README.md#initialdata) which can be used to load excalidraw with existing library items.

>   - Assign an [unique ID](https://github.com/excalidraw/excalidraw/blob/master/src/packages/excalidraw/README.md#Id) to the excalidraw component which can be accessed via [`ref`](https://github.com/excalidraw/excalidraw/blob/master/src/components/App.tsx#L265).

Reviewer: ```suggestion
  - Assign a [unique id](https://github.com/excalidraw/excalidraw/blob/master/src/packages/excalidraw/README.md#Id) to the excalidraw component. The id can be accessed via [`ref`](https://github.com/excalidraw/excalidraw/blob/master/src/components/App.tsx#L265).
```

  #### BREAKING CHANGE

>   - Earlier since there was no API to control the library hence was using local storage in the consumer. Now onwards with above API support the consumer will be responsible for handling the storage for the library.

Reviewer: ```suggestion
  - From now on the host application is responsible for [persisting the library]((https://github.com/excalidraw/excalidraw/blob/master/src/packages/excalidraw/README.md#onLibraryChange) to LocalStorage (or elsewhere), and [importing it on mount]((https://github.com/excalidraw/excalidraw/blob/master/src/packages/excalidraw/README.md#initialdata).
```
